pkg/controller/core/workload_controller.go

```diff
@@ -194,16 +187,39 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if updated {
 			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
 		}
-	} else if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
-		// if job is not active and does not have condition reason of WorkloadEvictedByDeactivation, update its condition
-		workload.SetEvictedCondition(&wl, kueue.WorkloadEvictedByDeactivation, "The workload is deactivated")
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
-			return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
+	} else {
+		var updated, evicted bool
```

Comment:
Maybe it is better to declare `evicted` as `*string` as it is used to bump the metric. 

I worry that if we have a `bool`, then someone can forget updating line 219 when adding a new scenario for deactivation. WDYT?

Reply:
@mimowo Uhm, do you assume that the `evicted` has the reason for eviction? (here is `InactiveWorkload`)?

Reply:
Oh, never mind, I thought for another scenario that Pods ready, but in that case it would still be WorkloadInactive

Reply:
Yeah, after we introduce a dedicated reason, we may refacotr here somehow.